src/Collapse/include/gudhi/Flag_complex_sparse_matrix.h

```diff
@@ -0,0 +1,424 @@
+      rows_++;
+    }
+  }
+
+  // Insert an edge in the data structure
+  void insert_new_edges(Vertex_handle u, Vertex_handle v, Filtration_value filt_val)
+  {
+    // The edge must not be added before, it should be a new edge.
+    insert_vertex(u, filt_val);
+    if (u != v) {
+      insert_vertex(v, filt_val);
+#ifdef DEBUG_TRACES
+      std::cout << "Insertion of the edge begins " << u <<", " << v << std::endl;
+#endif  // DEBUG_TRACES
+
+      auto rw_u = vertex_to_row_.find(u);
+      auto rw_v = vertex_to_row_.find(v);
+#ifdef DEBUG_TRACES
+      std::cout << "Inserting the edge " << u <<", " << v << std::endl;
+#endif  // DEBUG_TRACES
+      sparse_row_adjacency_matrix_.insert(rw_u->second, rw_v->second) = filt_val;
+      sparse_row_adjacency_matrix_.insert(rw_v->second, rw_u->second) = filt_val;
+    }
+#ifdef DEBUG_TRACES
+    else {
+     	std::cout << "Already a member simplex,  skipping..." << std::endl;
+    }
+#endif  // DEBUG_TRACES
+  }
+
+ public:
+  /** \brief Flag_complex_sparse_matrix constructor from a range of filtered edges.
+   *
+   * @param[in] filtered_edge_range Range of filtered edges. Filtered edges must be in
+   * `Flag_complex_sparse_matrix::Filtered_edge`.
+   *
+   * There is no need the range to be sorted, as it will be performed in
+   * `Flag_complex_sparse_matrix::filtered_edge_collapse`.
+   */
+  template<typename Filtered_edge_range>
+  Flag_complex_sparse_matrix(const Filtered_edge_range& filtered_edge_range)
+  : f_edge_vector_(filtered_edge_range.begin(), filtered_edge_range.end()),
+    rows_(0) {
+    for (Filtered_edge filtered_edge : filtered_edge_range) {
+      Vertex_handle u;
+      Vertex_handle v;
+      std::tie(u,v) = std::get<0>(filtered_edge);
+      vertices_.emplace(u);
+      vertices_.emplace(v);
+    }
+  }
+
+  /** \brief Flag_complex_sparse_matrix constructor from a proximity graph, cf. `Gudhi::compute_proximity_graph`.
+   *
+   * @param[in] one_skeleton_graph The one skeleton graph. The graph must be in
+   * `Flag_complex_sparse_matrix::Proximity_graph`.
+   *
+   * The constructor is computing and filling a vector of `Flag_complex_sparse_matrix::Filtered_edge`
+   */
+  Flag_complex_sparse_matrix(const Proximity_graph& one_skeleton_graph)
+  : rows_(0) {
+    // Insert all vertices_
+    for (auto v_it = boost::vertices(one_skeleton_graph); v_it.first != v_it.second; ++v_it.first) {
+      vertices_.emplace(*(v_it.first));
+    }
+    // Insert all edges
+    for (auto edge_it = boost::edges(one_skeleton_graph);
+         edge_it.first != edge_it.second; ++edge_it.first) {
+      auto edge = *(edge_it.first);
+      Vertex_handle u = source(edge, one_skeleton_graph);
+      Vertex_handle v = target(edge, one_skeleton_graph);
+      f_edge_vector_.push_back({{u, v}, boost::get(Gudhi::edge_filtration_t(), one_skeleton_graph, edge)});
+    }
+  }
+
+  /** \brief Performs edge collapse in a decreasing sequence of the filtration value.
```

Comment:
The user manual says increasing?

Reply:
Fix on 4fdc721. I checked on user manual and it is increasing.